SecurityService.OpenIdConnect.IntegrationTests/Common/DockerHelper.cs

```diff
@@ -1,137 +1,332 @@
+
+            this.Containers.AddRange(new List<IContainerService>
+                                     {
+                                         securityServiceContainer,
+                                         securityServiceTestUIContainer
+                                     });
         }
 
-        public async Task StopContainersForScenarioRun()
+        /// <summary>
+        /// Stops the containers for scenario run.
+        /// </summary>
+        public override async Task StopContainersForScenarioRun()
         {
-            try
+            if (this.Containers.Any())
             {
-                if (this.SecurityServiceTestUIContainer != null)
+                foreach (IContainerService containerService in this.Containers)
                 {
-                    this.SecurityServiceTestUIContainer.StopOnDispose = true;
-                    this.SecurityServiceTestUIContainer.RemoveOnDispose = true;
-                    this.SecurityServiceTestUIContainer.Dispose();
+                    containerService.StopOnDispose = true;
+                    containerService.RemoveOnDispose = true;
+                    containerService.Dispose();
                 }
+            }
 
-                if (this.TestNetwork != null)
+            if (this.TestNetworks.Any())
+            {
+                foreach (INetworkService networkService in this.TestNetworks)
                 {
-                    this.TestNetwork.Stop();
-                    this.TestNetwork.Remove(true);
+                    networkService.Stop();
+                    networkService.Remove(true);
                 }
             }
-            catch (Exception e)
+        }
+
+        /// <summary>
+        /// Adds the entry to hosts file.
+        /// </summary>
+        /// <param name="ipaddress">The ipaddress.</param>
+        /// <param name="hostname">The hostname.</param>
+        private static void AddEntryToHostsFile(String ipaddress,
+                                                String hostname)
+        {
+            if (FdOs.IsWindows())
             {
-                Console.WriteLine(e);
+                using(StreamWriter w = File.AppendText(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.System), @"drivers\etc\hosts")))
+                {
+                    w.WriteLine($"{ipaddress} {hostname}");
+                }
+            }
+            else if (FdOs.IsLinux())
+            {
+                DockerHelper.ExecuteBashCommand($"echo {ipaddress} {hostname} | sudo tee -a /etc/hosts");
             }
         }
-    }
 
-    [Binding]
-    public class Hooks
-    {
-        private readonly IObjectContainer ObjectContainer;
-        private IWebDriver WebDriver;
+        /// <summary>
+        /// Executes the bash command.
+        /// </summary>
+        /// <param name="command">The command.</param>
+        /// <returns></returns>
+        private static String ExecuteBashCommand(String command)
```

Comment:
Codacy found an issue: [Change return type to 'void'; not a single caller uses the returned value.](https://app.codacy.com/manual/stuart_ferguson1/SecurityService/pullRequest?prid=6144581)